packages/runner-ct/src/lib/event-manager.js

```diff
@@ -195,6 +196,17 @@ const eventManager = {
       this._clearAllCookies()
     })
 
+    const hmrSocket = connectWebpackHmr({
+      url: `${window.location.origin}/cypress-webpack-hmr-socket`,
+      onReload: () => {
+        if (state.isInitialBuildSucceed) {
+          rerun()
+        } else {
+          state.initialBuildFired()
+        }
+      },
+    })
+
```

Comment:
Will need to refactor this to be webpack agnostic at some point. I added a ticket to Jira.